devimint/src/external.rs

```diff
@@ -240,13 +241,54 @@ impl Bitcoind {
         Ok(proof.encode_hex())
     }
 
-    pub async fn get_raw_transaction(&self, txid: bitcoin30::Txid) -> Result<String> {
+    /// Get a transaction by its txid. Checks the mempool and all blocks.
+    pub async fn get_transaction(&self, txid: bitcoin30::Txid) -> Result<String> {
+        // Check the mempool.
+        if let Ok(tx) = self.get_raw_transaction(txid, None).await {
+            return Ok(tx);
+        };
+
+        let block_height = self.get_block_count().await? - 1;
+
+        // Check each block for the tx, starting at the chain tip.
+        // Buffer the requests to avoid spamming bitcoind.
+        // We're doing this after checking the mempool since the tx should
+        // usually be in the mempool, and we don't want to needlessly hit
+        // the bitcoind with block requests.
+        let mut buffered_tx_stream = futures::stream::iter((0..block_height).rev())
+            .map(|height| async move {
+                let block_hash = self.get_block_hash(height).await?;
+                let tx = self.get_raw_transaction(txid, Some(block_hash)).await?;
+                Ok::<_, anyhow::Error>(tx)
+            })
+            .buffered(32);
+
+        while let Some(tx_or) = buffered_tx_stream.next().await {
```

Comment:
Do you need a loop here? Should one be one transaction right?

Reply:
Oh, actually, shouldn't we be retrying on errors here anyway?

Reply:
I could imagine polling the mempool while scanning the chain as a performance improvement, but fundamentally the caller needs to re-try if nothing is found.

Reply:
Well, the function is called `get_transaction`, and the caller might have a reason to know that the tx is already there, etc.

Anything OK, as long as we don't swallow errors.

Reply:
I changed this so we can now differentiate between the RPC call failing vs the transaction not being found. We now return an error if an RPC fails but continue checking more blocks if the transaction isn't found in a given block.